### PR TITLE
Silence declared but unused warnings when there is no loggging code g…

### PR DESCRIPTION
**Pull request by jangko**
…enerated

### CODE REVIEW

File: chronicles.nim [hunk 1]
```diff
@@ -216,14 +216,18 @@ macro expandIt*(T: type, expandedProps: untyped): untyped =
   when defined(debugLogImpl):
     echo result.repr
 
+template chroniclesUsedMagic(x: untyped) =
+  # Force the compiler to mark any symbol in the x
+  # as used without actually generate any code.
+  when compiles(x): discard
```
Comment (arnetheduck) [inline chronicles.nim:222]: can't they be annotated with `{.used.}` instead? this feels like it will further add to compilation slowness..

Comment (jangko) [reply chronicles.nim:222]: Not always, for example:

```Nim
try:
   ...
except SomeError as exc:   # cannot use {.used.} here
   debug "Some text", msg=exc.msg
```

I don't know when we will get this feature in next nim version. Until then this is the available workaround.

Comment (arnetheduck) [reply chronicles.nim:222]: is there an upstream bug?

Comment (jangko) [reply chronicles.nim:222]: It simply not exists

**Pull request merged**